### PR TITLE
[Behat] Checked if the shipment confirmation email gets sent

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | yes
| Related tickets | - 
| License         | MIT

Based on #5912

### CODE REVIEW

File: features/account/registration.feature [hunk 1]
```diff
@@ -47,3 +47,9 @@ Feature: Account registration
         Then I should be notified that new account has been successfully created
         And I should be logged in
         And my email should be "goodman@gmail.com"
+
+    @ui @email
+    Scenario: Receiving a welcoming email after registration
+        When I register with email "ghastly@bespoke.com" and password "suitsarelife"
+        Then I should be notified that new account has been successfully created
```
Comment (TheMadeleine) [inline features/account/registration.feature]: **a** new account

Comment (NeverResponse) [inline features/account/registration.feature]: I'd go with `my account` as it's an `I register` scenario. WDYT?

File: features/order/managing_orders/shipping_order.feature [hunk 2]
```diff
@@ -15,12 +15,13 @@ Feature: Shipping an order
...
 
-    @ui
+    @ui @email
     Scenario: Finalizing order's shipment
         Given I view the summary of the order "#00000666"
         When specify its tracking code as "#00044"
         And I ship this order
-        Then I should be notified that the order's shipment has been successfully shipped
+        Then I should be notified that the order has been successfully shipped
+        And an email with shipment's details of order "#00000666" should be sent to "lucy@teamlucifer.com"
```
Comment (michalmarcinkowski) [inline features/order/managing_orders/shipping_order.feature]: ... of **this** order ~~"#00000666"~~ ...

File: src/Sylius/Behat/Context/Ui/EmailContext.php [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * @Then :count email(s) should be sent to :recipient
+     */
+    public function numberOfEmailsShouldBeSentTo($count, $recipient)
+    {
+        $this->assertEmailHasRecipient($recipient);
+
+        Assert::eq(
+            $this->emailChecker->getMessagesCount(),
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Ui/EmailContext.php]: We should check if X messages were send to the specific recipient, not only that X messages were send.

File: src/Sylius/Behat/Context/Ui/EmailContext.php [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+        );
+    }
+
+    /**
+     * @Then a welcoming email should have been sent to :recipient
+     */
+    public function aWelcomingEmailShouldHaveBeenSentTo($recipient)
+    {
+        $this->assertEmailHasRecipient($recipient);
+        $this->assertEmailContainsMessage('Welcome to our store');
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Ui/EmailContext.php]: We should perform a deeper check and not only validate some email with this message was sent but also that this message was sent to the specific recipient `assertEmailToRecipientContainsMessage($recipient, 'Welcome to our store')`

File: src/Sylius/Behat/Context/Ui/EmailContext.php [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+     */
+    public function anEmailWithShipmentDetailsOfOrderShouldBeSentTo(OrderInterface $order, $recipient)
+    {
+        $this->assertEmailHasRecipient($recipient);
+
+        $this->assertEmailContainsMessage($order->getNumber());
+        $this->assertEmailContainsMessage($order->getLastShipment()->getMethod()->getName());
+
+        $tracking = $this->sharedStorage->get('tracking_code');
+        $this->assertEmailContainsMessage($tracking);
```
Comment (michalmarcinkowski) [inline src/Sylius/Behat/Context/Ui/EmailContext.php]: Maybe we could squash it to the single check? `Your order %number% has been shipped using %method%. You can check it's location with the "%tracking%" tracking code.`

**Comment by michalmarcinkowski** [comment]
Nice work Janek! Please apply my comments in a separate PR.

**Pull request merged**